core/store/src/trie/mem/flexible_data/children.rs

```diff
@@ -0,0 +1,99 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+
+use super::encoding::BorshFixedSize;
+use super::FlexibleDataHeader;
+use crate::trie::mem::arena::{ArenaSlice, ArenaSliceMut};
+use crate::trie::mem::node::{MemTrieNodeId, MemTrieNodePtr};
+use crate::trie::Children;
+
+/// Flexibly-sized data header for a variable-sized list of children trie nodes.
+/// The header contains a 16-bit mask of which children are present, and the
+/// flexible part is one pointer for each present child.
+#[derive(Clone, Copy, BorshSerialize, BorshDeserialize)]
+pub struct EncodedChildrenHeader {
+    mask: u16,
+}
+
+impl BorshFixedSize for EncodedChildrenHeader {
+    const SERIALIZED_SIZE: usize = std::mem::size_of::<u16>();
```

Comment:
Nit: we ideally need fuzzing for each and every implementation of `BorshFixedSize`.

Maybe make a `borsh_fixed_size!` macro that requires the type to implement `TypeGenerator`, and at the same time implements `BorshFixedSize` and a fuzzer test that validates the length of the borsh encoding?

Reply:
Yeah... I didn't want to overengineer this coz higher level test coverages would easily find holes here (if a size doesn't match, the RawEncoder would complain, and during decoding we would just get some garbage on the next struct.)

Ideally borsh crate would provide this feature - it doesn't even have to be static. If BorshSerialize could just provide a serialized_size<T: BorshSerialize>(data: &T), that would be good enough.

Reply:
It actually does provide the equivalent of a `serialized_size`, but it takes roughly as long as serializing the whole thing (unless rustc is very good at optimizing)

Borsh allows you to serialize to any `Write` struct. So it’s possible to just make a struct implementing `Write` that only ever counts the number of bytes written to the sink. That said, given the probable runtime of all the things, unless the serialized data is very large, it’s likely going to be faster to just serialize to a Vec and then copy out of it.

Reply:
I saw that the latest borsh has this capability now, and I did try (after this PR) to see if I could get rid of this fixed size thing. But it actually became a bit awkward because during encoding, we don't know the encoded size. If we don't know the encoded size, we would basically need to pass in a slice that goes from wherever we're decoding from to the very end of the whole arena, which is just absurd.

Reply:
👌 then I guess the macro I suggested below to auto-generate both the implementation and tests would be the solution then